### ISSUE TITLE
[Feature Pitch] Tweaks to Kegs.

### ISSUE THREAD

**Issue by Cruxiat** [transhumandesign/kag-base#279]
- water buckets deactivate lit kegs
- fire and fire arrows activate kegs

having some small counter-play within the way kegs are played currently would be great for introducing more strategy into the way kegs are currently played with. Currently, if a keg is lit and thrown from a tower the only way to defend from it is either a lucky catch or just running and hoping you don't die. Having water buckets (an object that can be thrown for a ranged splash) means that teams will have to work together & focus on using kegs to the fullest potential; ie; playing as a team. 

I chose buckets because having builders be the ones to defuse kegs seems to me like the most balanced way of implementing this feature. 

The fire arrows on the other hand serve as ways to punish teams for leaving kegs out in the open (or dropping them) which encourages the careful use of kegs (as opposed to just spamming them / blindly running in with them) 

Hopefully these changes encourage smart thinking, teamwork, and above all better quality matches.

**Comment by Franrek** [comment]
Bucket meta lets go.

**Comment by Vikhenzo** [comment]
Fire lighting them seems too much, water in general deactivating the wick seems fine though

**Comment by benjamingorman** [comment]
Actually it's possible to throw a water bomb at a falling keg and thus bounce it back at the thrower. It's just really hard to time properly.

The bucket idea is nice though the timing window to stop falling kegs would be really tight. It's still great for kegs on the ground though and would encourage builders to keep a bucket handy.

**Comment by terracraft321** [comment]
Can we keep this realistic, without ruining the gameplay?

**Comment by FuzzCat927** [comment]
> Can we keep this realistic, without ruining the gameplay?

"realistic"
If you're trying to argue that kag is realistic, lol.
otherwise, mind explaning what you mean?

**Comment by CatWiz** [comment]
If water buckets will deactivate kegs, but water arrows/bombs will not, it wont make any sense. If they will, kegs will be practically unusable.

> I chose buckets because having builders be the ones to defuse kegs seems to me like the most balanced way of implementing this feature.

i think you forgot that archers and knights can use water buckets as well.

>The fire arrows on the other hand serve as ways to punish teams for leaving kegs out in the open (or dropping them) which encourages the careful use of kegs (as opposed to just spamming them / blindly running in with them)

losing a keg in battle already punishes team who lost it a lot, and archers already can already steal enemy kegs if they are placed badly. allowing them to light it right where they took it would give enemy team almost no chance to counter it, since archers, unlike knights, can climb 40 block tower in few seconds to drop keg from above. of course you can catch it, but average player just cant do that. imagine good archers pubstorming by carrying kegs to enemy base and launching them into the walls. and you wont even be able to defuse it, since they will instantly explode.  doesnt seem like quality match to me

tldr archers already op dont buff them

**Comment by eps0003** [comment]
> If water buckets will deactivate kegs, but water arrows/bombs will not, it wont make any sense.
I don't see what's wrong with giving a bucket a unique function. If all water items defuse kegs, that'll make launching kegs with water bombs impossible. We can't prevent defusing team kegs since that'll be unintuitive because kegs aren't a team explosive.

**Comment by 1bardesign** [comment]
This pitch has a **$20 bounty**, but it has a PR waiting ( #298 ) - starting new work on it is not recommended.

**Comment by ParaLogia** [comment]
@benjamingorman , filled water buckets can be bought at builder shops. I often see people using this feature, when they're trying to defend a wooden skybridge.
